src/Microsoft.Identity.Client/Internal/IdToken.cs

```diff
@@ -73,22 +73,33 @@ internal class IdToken
 
         public static IdToken Parse(string idToken)
         {
+            if (string.IsNullOrEmpty(idToken))
+            {
+                return null;
+            }
+
             IdToken idTokenBody = null;
-            if (!string.IsNullOrWhiteSpace(idToken))
+            string[] idTokenSegments = idToken.Split(new[] {'.'});
+
+            if (idTokenSegments.Length < 2)
             {
-                string[] idTokenSegments = idToken.Split(new[] {'.'});
+                throw new MsalClientException(MsalClientException.InvalidJwtError, "ID Token must contain at least 2 parts.");
```

Comment:
Per this issue https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/93, shouldn't change it to MsalServiceException?